inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -151,133 +148,171 @@ void MKLDNNPadNode::initSupportedPrimitiveDescriptors() {
-        padsEnd[1] /= params.srcDims[params.srcDims.size() - 1];
-        padsBegin.push_back(0);
-        padsEnd.push_back(0);
+    // pads are constant, so we can calculate new collapsing pads for first target dimensions and use it for the next dimensions
+    // to avoid permanent identical pad calculations
+    if (srcMemPtr->getDesc().hasLayoutType(LayoutType::nCsp16c)) {
+        attrs.padsBegin[1] /= 16;
+        attrs.padsEnd[1] /= 16;
+        attrs.padsBegin.push_back(0);
+        attrs.padsEnd.push_back(0);
+    } else if (srcMemPtr->getDesc().hasLayoutType(LayoutType::nCsp8c)) {
+        attrs.padsBegin[1] /= 8;
+        attrs.padsEnd[1] /= 8;
+        attrs.padsBegin.push_back(0);
+        attrs.padsEnd.push_back(0);
     } else {
-        auto order = inBlkDesc->getOrder();
-        std::vector<unsigned int> newPadsBegin(padsBegin.size(), 0), newPadsEnd(padsEnd.size(), 0);
-        for (size_t i = 0; i < padsBegin.size(); ++i) {
-            newPadsBegin[i] = padsBegin[order[i]];
-            newPadsEnd[i] = padsEnd[order[i]];
+        auto order = getParentEdgeAt(0)->getMemory().GetDescWithType<BlockedMemoryDesc>()->getOrder();
+        std::vector<unsigned int> newPadsBegin(attrs.padsBegin.size(), 0), newPadsEnd(attrs.padsEnd.size(), 0);
+        for (size_t i = 0; i < attrs.padsBegin.size(); ++i) {
+            newPadsBegin[i] = attrs.padsBegin[order[i]];
+            newPadsEnd[i] = attrs.padsEnd[order[i]];
         }
-        padsBegin = newPadsBegin;
-        padsEnd = newPadsEnd;
+        attrs.padsBegin = newPadsBegin;
+        attrs.padsEnd = newPadsEnd;
     }
 
-    int beginIdx = 0;
-    int endIdx = padsBegin.size() - 1;
+    // collapse dimensions
+    attrs.beginPadIdx = 0;
+    attrs.endPadIdx = attrs.padsBegin.size() - 1;
 
-    for (int i = 0; i < padsBegin.size(); ++i) {
-        if (padsBegin[i] != 0 || padsEnd[i] != 0) {
-            beginIdx = i - 1;
+    for (int i = 0; i < attrs.padsBegin.size(); ++i) {
+        if (attrs.padsBegin[i] != 0 || attrs.padsEnd[i] != 0) {
+            attrs.beginPadIdx = i - 1;
             break;
         }
     }
 
-    for (int i = padsBegin.size() - 1; i >= 0; --i) {
-        if (padsBegin[i] != 0 || padsEnd[i] != 0) {
-            endIdx = i;
+    for (int i = attrs.padsBegin.size() - 1; i >= 0; --i) {
+        if (attrs.padsBegin[i] != 0 || attrs.padsEnd[i] != 0) {
+            attrs.endPadIdx = i;
             break;
         }
     }
 
-    params.lastDstDim = params.dstStrides[std::max(endIdx - 1, 0)];
-    params.nDimsForWork = endIdx - std::max(beginIdx, 0);
+    if (attrs.beginPadIdx > 0) {
+        attrs.padsBegin.erase(attrs.padsBegin.begin() + 1, attrs.padsBegin.begin() + attrs.beginPadIdx + 1);
+        attrs.padsEnd.erase(attrs.padsEnd.begin() + 1, attrs.padsEnd.begin() + attrs.beginPadIdx + 1);
+    }
+
+    attrs.prc = getSelectedPrimitiveDescriptor()->getConfig().inConfs[0].desc->getPrecision();
```

Comment:
Are there any reasons why this is not good? Because we use memory pointer in this method anyway so using this pointer is better? I changed but I'd to know reason for future.

Reply:
Of course precision on the port config and  precision of the tensor descriptor in the memory  must be the same it is guarantied by the common graph logic. But, as far as I can understand the executor works not with port config but with memory and tensor descriptor that belongs to that memory object, so it looks more logical to read this data from the actual entity that the executor work with. Moreover, it is strange to query config when you have memory pointer under your hand.